CMakeLists.txt

```diff
@@ -138,6 +138,7 @@ sls_enable_cxx_warning("-Wduplicated-cond")
 sls_disable_cxx_warning("-Wclass-memaccess")
 sls_disable_c_warning("-Wstringop-truncation")
 sls_disable_c_warning("-Wformat-nonliteral")
+sls_disable_c_warning("-Wformat-truncation")
```

Comment:
In principle we don't need to disable this warning. It only warns if the formatting can be truncated and this is not acted on. 

Code like this doesn't generate a warning:
```
int r = snprintf(mess, MAX_STR_LENGTH-100, "Could not get kernel version. %s\n",
                 retvals);
if (r>MAX_STR_LENGTH)
     mess[MAX_STR_LENGTH-2] = '.';
```

In principle we could have a function that adds "..." when the string is truncated. Will it happen? Is it important? Nor sure =)